examples/src/main/kotlin/ai/koog/agents/example/chess/choice/ChessChoiceNodes.kt

```diff
@@ -0,0 +1,81 @@
+package ai.koog.agents.example.chess.choice
+
+import ai.koog.agents.core.agent.AIAgent
+import ai.koog.agents.core.dsl.builder.forwardTo
+import ai.koog.agents.core.dsl.builder.strategy
+import ai.koog.agents.core.dsl.extension.nodeExecuteTool
+import ai.koog.agents.core.dsl.extension.nodeLLMRequest
+import ai.koog.agents.core.dsl.extension.onAssistantMessage
+import ai.koog.agents.core.dsl.extension.onToolCall
+import ai.koog.agents.core.environment.ReceivedToolResult
+import ai.koog.agents.core.feature.choice.AskUserChoiceStrategy
+import ai.koog.agents.core.feature.choice.nodeChoose
+import ai.koog.agents.core.feature.choice.nodeLLMSendResultsMultipleChoices
+import ai.koog.agents.core.tools.ToolRegistry
+import ai.koog.agents.example.ApiKeyService
+import ai.koog.agents.example.chess.ChessGame
+import ai.koog.agents.example.chess.Move
+import ai.koog.agents.example.chess.nodeTrimHistory
+import ai.koog.prompt.executor.clients.openai.OpenAIModels
+import ai.koog.prompt.executor.llms.all.simpleOpenAIExecutor
+import ai.koog.prompt.message.Message
+import kotlinx.coroutines.runBlocking
+
+fun main() = runBlocking {
+    val game = ChessGame()
+
+    val toolRegistry = ToolRegistry {
+        tools(listOf(Move(game)))
+    }
+
+    val askChoiceStrategy = AskUserChoiceStrategy(promptShowToUser = { prompt ->
+        val lastMessage = prompt.messages.last()
+        if (lastMessage is Message.Tool.Call) {
+            lastMessage.content
+        } else {
+            ""
+        }
+    })
+
+    val strategy = strategy("chess_strategy") {
+        val nodeCallLLM by nodeLLMRequest("sendInput")
+        val nodeExecuteTool by nodeExecuteTool("nodeExecuteTool")
+        val nodeSendToolResult by nodeLLMSendResultsMultipleChoices("nodeSendToolResult")
+        val nodeChooseChoice by nodeChoose(askChoiceStrategy, "chooseLLMChoice")
```

Comment:
Let's rename "nodeChoose" to something more verbose like "nodeChooseLLMResponse"

Reply:
I mean in the public API also